platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -323,14 +352,76 @@ - (MGLImage *)blurredAttributionBackground:(CIImage *)backgroundImage
-    
     return [UIImage imageWithCGImage:cgimg];
 #else
-    
     return [[NSImage alloc] initWithCGImage:cgimg size:[backgroundImage extent].size];
 #endif
 }
 
+- (MGLSnapshotAttributionOptions *)attributionOptionsForSize:(CGSize)snapshotSize attributionInfo:(NSArray *)attributionInfo
+{
+    NSMutableArray *options = [NSMutableArray array];
+    MGLSnapshotAttributionOptions *largeLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    largeLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    largeLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox" ofType:@"pdf"]];
+#endif
+    _defaultLogoHeight = largeLogoAttribution.logoImage.size.height;
+    
+    MGLSnapshotAttributionOptions *smallLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    smallLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox_helmet" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    smallLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox_helmet" ofType:@"pdf"]];
+#endif
+    
+    MGLSnapshotAttributionOptions *noLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:YES];
+    
+    [options addObject:largeLogoAttribution];
+    [options addObject:smallLogoAttribution];
+    [options addObject:noLogoAttribution];
+    
+    for (MGLSnapshotAttributionOptions *attributionOptions in options) {
+        // -[Mapbox Logo]-[Attribution Background]-
+        CGFloat origin = attributionOptions.logoImage ? MGLLogoImagePosition.x : 0;
+        CGFloat width = origin + attributionOptions.logoImage.size.width + 10 + attributionOptions.attributionBackgroundSize.width + 10;
+        if (width <= snapshotSize.width) {
+            return attributionOptions;
+        }
+    }
+    
+    return noLogoAttribution;
+}
+
+- (MGLSnapshotAttributionOptions *)attributionOptionsForAttributionInfo:(NSArray *)attributionInfo abbreviated:(BOOL)abbreviated
+{
+    NSString *openStreetMap = NSLocalizedStringWithDefaultValue(@"OPEN_STREET_MAP_ATTRIBUTION", nil, nil, @"OpenStreetMap", @"Open Street Map full name attribution");
+    NSString *OSM = NSLocalizedStringWithDefaultValue(@"OSM_ATTRIBUTION", nil, nil, @"OSM", @"Open Street Map short name attribution");
+    NSMutableArray *infos = [NSMutableArray array];
+    CGSize attributionBackgroundSize = CGSizeMake(10, 0);
+    for (MGLAttributionInfo *info in attributionInfo) {
+        if (info.isFeedbackLink) {
+            continue;
+        }
+        MGLAttributionInfo *attribution = [info copy];
+        if (abbreviated && [[info.title string] rangeOfString:openStreetMap].location != NSNotFound) {
```

Comment:
The string we’re looking for is “OpenStreetMap”, unlocalized, because the most common OpenStreetMap-based sources have this string baked in (mapbox/tilejson-spec#20). If we see “OpenStreetMap” (unlocalized), we should replace it with “OpenStreetMap” (localized) or “OSM” (localized) depending on `isAbbreviated`.